### PR TITLE
Charms should take their series from the bundle if needed

### PR DESCRIPTION
**Pull request by wallyworld**
Fixes https://bugs.launchpad.net/juju/+bug/1796378

When determining the series for a charm, if non were specified for the application, the charm default was being used (the first declared series in the charm metadata). Instead, we should have been using the bundle default, if any.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/bundlechanges/151/

File: handlers.go [hunk 1]
```diff
@@ -957,22 +957,25 @@ func placeholder(changeID string) string {
 	return "$" + changeID
 }
 
-func isNewMachine(id string) bool {
```
Comment (Veebers) [inline handlers.go]: I presume this is a driveby fix?

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/bundlechanges-merge
